de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/CustomDialog.java

```diff
@@ -0,0 +1,377 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.uiengine.ui.editor.snippets.general;
+
+import java.util.Arrays;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.List;
+import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.swt.widgets.Monitor;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+
+/**
+ * CustomDialog is a custom dialog that extends the Dialog class, allowing user interaction to modify a list of users.
+ * The dialog provides options to add, remove, update, and rearrange users in the list. It also notifies the parent application
+ * of user changes using a callback mechanism.
+ *
+ * This dialog is primarily used to interact with and manage a list of features displayed to the user, enabling them to make necessary modifications.
+ *
+ * This dialog includes the following features:
+ * - Adding a new feature to the list
+ * - Removing an existing feature from the list
+ * - Updating an existing feature in the list
+ * - Moving a feature up or down in the list
+ *
+ * The dialog is configured to have a specified size and is centered on the screen for user convenience.
+ *
+ * Usage:
+ * 1. Create an instance of CustomDialog, providing the parent shell, initial features to display, and a callback for feature updates.
+ * 2. Open the dialog using the open() method.
+ *
+ * @author ngat_di
+ */
+
+public class CustomDialog extends Dialog {
+
+    // Fields
+    private String[] features;  // Array of features
+    private List featuresList;  // SWT List for displaying features
+    private FeatureUpdateCallback featureUpdateCallback;  // Callback for updating features
+
+    /**
+     * Constructor for CustomDialog.
+     *
+     * @param parentShell           Parent shell for the dialog
+     * @param initialFeatures       Initial features to display
+     * @param callback              Callback to handle feature updates
+     */
+
+    public CustomDialog(Shell parentShell, String[] initialFeatures, FeatureUpdateCallback callback) {
+        super(parentShell);
```

Comment:
I think it might be cool to have a title on this dialog... (Which is shown on top of this window) I think there is a method setTitle(...) or so?

Reply:
I gave the title User--Discipline

Reply:
The method is setText() and i did the change on configureShell(Shell newShell) method.